app/assets/javascripts/spree/frontend/solidus_paypal_commerce_platform/button_actions.js

```diff
@@ -11,6 +11,56 @@ SolidusPaypalCommercePlatform.postOrder = function(payment_method_id) {
   })
 }
 
+SolidusPaypalCommercePlatform.createAndSendOrder = function(payment_method_id) {
+  return SolidusPaypalCommercePlatform.createOrder().then(function(){
+    return SolidusPaypalCommercePlatform.sendOrder(payment_method_id)
+  })
+}
+
+SolidusPaypalCommercePlatform.createOrder = function() {
+  var data = {
+    order: {
+      line_items_attributes: [{
+        variant_id: SolidusPaypalCommercePlatform.getVariantId(),
+        quantity: SolidusPaypalCommercePlatform.quantity_field.value
+      }]
+    }
+  }
+
+  if(typeof SolidusPaypalCommercePlatform.user_id !== 'undefined'){
+    data["user_id"] = SolidusPaypalCommercePlatform.user_id
+  }
+
+  return Spree.ajax({
+    url: "/solidus_paypal_commerce_platform/orders",
+    method: 'POST',
+    data: data,
+    error: function(response) {
+      message = response.responseJSON
+      alert('A problem has occurred while creating your order - ' + message);
+    }
+  }).then(function(response) {
+    Spree.current_order_id = response.number
+    Spree.current_order_token = response.guest_token
+  });
+}
+
+SolidusPaypalCommercePlatform.getVariantId = function() {
+  var variants = SolidusPaypalCommercePlatform.variants
```

Comment:
I think the value of `.variants` can be inlined. If I'm understanding correctly the intention, we can still allow customization by having `.getVariantId` replaced if the way it's retrieved is different for a given store.

Reply:
Yeah, that makes more sense for customization. Thanks!